### PR TITLE
Fix unwrap and lock_account

### PR DESCRIPTION
**Pull request by Index0011**
fix unwrap and lock_account

### CODE REVIEW

File: pallets/phala/src/compute/wrapped_balances.rs [hunk 1]
```diff
@@ -268,12 +269,17 @@ pub mod pallet {
 				Error::<T>::UnwrapAmountExceedsAvaliableStake
 			);
 			let active_stakes = Self::get_net_value(user.clone())?;
-			let staker_status =
-				StakerAccounts::<T>::get(&user).ok_or(Error::<T>::StakerAccountNotFound)?;
-			ensure!(
-				amount + staker_status.locked <= active_stakes,
-				Error::<T>::UnwrapAmountExceedsAvaliableStake,
-			);
+			if let Some(staker_status) = StakerAccounts::<T>::get(&user) {
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/wrapped_balances.rs:272]: Better like this?

```
let locked = StakerAccounts::<T>::get(user).map_or(0, |status| status.locked);
```

File: pallets/phala/src/compute/wrapped_balances.rs [hunk 2]
```diff
@@ -240,9 +240,10 @@ pub mod pallet {
 			let free_stakes: BalanceOf<T> = <pallet_assets::pallet::Pallet<T> as Inspect<
 				T::AccountId,
 			>>::balance(T::WPhaAssetId::get(), &user);
-			let staker_status =
-				StakerAccounts::<T>::get(&user).ok_or(Error::<T>::StakerAccountNotFound)?;
-			let withdraw_amount = (active_stakes - staker_status.locked).min(free_stakes);
+			let mut withdraw_amount = free_stakes;
+			if let Some(staker_status) = StakerAccounts::<T>::get(&user) {
+				withdraw_amount = (active_stakes - staker_status.locked).min(free_stakes);
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/wrapped_balances.rs:245]: Check "unwrap()"

File: pallets/phala/src/compute/stake_pool_v2.rs [hunk 3]
```diff
@@ -802,6 +802,67 @@ pub mod pallet {
...
+		pub fn reset_iter_pos(origin: OriginFor<T>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			base_pool::Pallet::<T>::ensure_migration_root(who)?;
+			StakepoolIterateStartPos::<T>::put(None::<u64>);
+			Ok(())
+		}
+
+		#[pallet::weight(0)]
+		#[frame_support::transactional]
+		pub fn migrate_stakepools(origin: OriginFor<T>, max_iterations: u32) -> DispatchResult {
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/stake_pool_v2.rs:816]: It's not the initial migration. Better to give it a name like "fix_missing_worker_lock()"

File: pallets/phala/src/compute/stake_pool_v2.rs [hunk 4]
```diff
@@ -802,6 +802,67 @@ pub mod pallet {
...
+			base_pool::Pallet::<T>::ensure_migration_root(who)?;
+			let mut last_pid = StakepoolIterateStartPos::<T>::get();
+			let mut iter = match last_pid {
+				Some(pid) => {
+					let key: Vec<u8> = base_pool::pallet::Pools::<T>::hashed_key_for(pid);
+					base_pool::pallet::Pools::<T>::iter_from(key)
+				}
+				None => base_pool::pallet::Pools::<T>::iter(),
+			};
+			let asset_account = <T as wrapped_balances::Config>::WPhaAssetId::get();
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/stake_pool_v2.rs:827]: It's the asset of, not an account

File: pallets/phala/src/compute/stake_pool_v2.rs [hunk 5]
```diff
@@ -802,6 +802,67 @@ pub mod pallet {
...
+						pool_info.cd_workers.into_iter().for_each(|pubkey| {
+							let session: T::AccountId = pool_sub_account(pid, &pubkey);
+							total_lock += computation::Stakes::<T>::get(&session).unwrap_or_default();
+						});
+						let curr_lock: BalanceOf<T> =
+						<pallet_assets::pallet::Pallet<T> as Inspect<T::AccountId>>::balance(
+							asset_account,
+							&pool_info.lock_account,
+						);
+						if curr_lock < total_lock {
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/stake_pool_v2.rs:847]: Is it possible curr_lock > total lock? For security, I suggest to either return an error here, or we can simulate it offline first.

Comment (Index0011) [reply pallets/phala/src/compute/stake_pool_v2.rs]: don e

File: pallets/phala/src/compute/stake_pool_v2.rs (hunk 4 above)
Comment (Index0011) [reply pallets/phala/src/compute/stake_pool_v2.rs]: done

File: pallets/phala/src/compute/stake_pool_v2.rs (hunk 3 above)
Comment (Index0011) [reply pallets/phala/src/compute/stake_pool_v2.rs]: done

File: pallets/phala/src/compute/wrapped_balances.rs (hunk 2 above)
Comment (Index0011) [reply pallets/phala/src/compute/wrapped_balances.rs]: done

File: pallets/phala/src/compute/wrapped_balances.rs (hunk 1 above)
Comment (Index0011) [reply pallets/phala/src/compute/wrapped_balances.rs]: done

**Review by h4x3rotab — APPROVED** [review]
LGTM

**Pull request merged**